src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -2,62 +2,160 @@ import { useCallback, useEffect, useState } from 'react'
+          <div className={styles.errorSubtitle}>
+            If this action was not initiated by you, feel free to dismiss this message and close this window.
+          </div>
+        </div>
+      </main>
+    )
+  }
+
+  if (resultError) {
+    return (
+      <main className={styles.main}>
+        <div className={styles.left}>
+          <div className={styles.errorLogo}></div>
+          <div className={styles.errorTitle}>An error ocurred while interacting with your wallet...</div>
+          <div className={styles.errorSubtitle}>Close this window and try again.</div>
+          <div className={styles.errorValue}>{resultError}</div>
+        </div>
+      </main>
+    )
+  }
+
+  if (resultSent) {
+    return (
+      <main className={styles.main}>
+        <div className={styles.left}>
+          <div className={styles.logo}></div>
+          <div className={styles.title}>Wallet Interaction Complete</div>
+          <div className={styles.description}>You can close this window now.</div>
+        </div>
+      </main>
+    )
   }
 
   if (!request) {
-    return <div style={{ margin: '1rem' }}>Loading...</div>
+    return (
+      <main className={styles.main}>
+        <div className={styles.left}>
+          <Loader active size="huge" />
+        </div>
+      </main>
+    )
   }
 
+  console.log(request)
+
   if (request.method === 'dcl_personal_sign') {
     return (
-      <div style={{ margin: '1rem' }}>
-        <h1>{request.method}</h1>
-        <pre>{request.params[0]}</pre>
-        <div>
-          Code: <b>{request.code}</b> (This code should be visible on the desktop client as well)
+      <main className={styles.main}>
+        <div className={styles.left}>
+          <div className={styles.logo}></div>
+          <div className={styles.title}>Verify Sign In</div>
+          <div className={styles.description}>Do you see the same verification number on your desktop app?</div>
+          <div className={styles.code}>{request.code}</div>
+          <div className={styles.buttons}>
+            <Button
+              className={styles.noButton}
+              onClick={() => {
+                setDenied(true)
+              }}
+            >
+              <div className={styles.noLogo}></div> No
+            </Button>
+            <Button
+              className={styles.yesButton}
+              onClick={async () => {
+                const provider = await connection.getProvider()
+                const browserProvider = new ethers.BrowserProvider(provider)
+                const signer = await browserProvider.getSigner()
+                const signature = await signer.signMessage(request.params[0])
+                const result = await authServerFetch('outcome', { requestId, sender: await signer.getAddress(), result: signature })
+
+                if (result.error) {
+                  setResultError(result.error)
+                }
+
+                setResultSent(true)
```

Comment:
Would you mind coding this function using `useCallback`?
The same would apply to the `onClick` method down below.